providers/tentacle.rb

```diff
@@ -129,6 +129,83 @@
   new_resource.updated_by_last_action(actions_updated?([install, create_home_dir, generate_cert, create_instance, configure, service]))
 end
 
+action :register do
+  new_resource = @new_resource
+  instance = new_resource.instance
+  port = new_resource.port
+  polling_port = new_resource.polling_port
+  polling = fancy_bool(new_resource.polling)
+  server = new_resource.server
+  api_key = new_resource.api_key
+  roles = new_resource.roles
+  environment = new_resource.environment
+
+  verify_server(server)
+  verify_api_key(api_key)
+  verify_roles(roles)
+  verify_environment(environment)
+
+  firewall = windows_firewall_rule 'Octopus Deploy Tentacle' do
```

Comment:
No so configuring a tentacle is a requirement to add it to the octo server.  I am simply saying that we should setup the firewall items in the configure section instead of in the register section.  In addition it should also be behind a flag.